pymodbus/client/asynchronous/factory/tcp.py

```diff
@@ -95,10 +94,10 @@ def async_io_factory(host="127.0.0.1", port=Defaults.Port, framer=None,
     proto_cls = kwargs.get("proto_cls", None)
     if not loop.is_running():
         asyncio.set_event_loop(loop)
-        cor = init_tcp_client(proto_cls, loop, host, port)
+        cor = init_tcp_client(proto_cls, loop, host, port, **kwargs)
         client = loop.run_until_complete(asyncio.gather(cor))[0]
     else:
-        cor = init_tcp_client(proto_cls, loop, host, port)
+        cor = init_tcp_client(proto_cls, loop, host, port, **kwargs)
```

Comment:
before this, remove from kwargs (i.e. {arg:argv[arg] for arg in kwargs if arg not in ["proto_cls","loop","host","port"]}